walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -11,13 +11,69 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.ArrayList;
+import java.util.List;
+// This class takes in a collection of events and a meeting request,
+// and generates potential times for the meeting. 
+public final class FindMeetingQuery { 
 
-public final class FindMeetingQuery {
+  ArrayList<TimeRange> viableTimeSlots = new ArrayList<TimeRange>();
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    int duration = (int)request.getDuration();
+
+    List<TimeRange> requiredBusyTimes = events.stream()
+      .filter(e -> eventContains(e.getAttendees(), request, false))
+      .map(e -> e.getWhen())
+      .collect(Collectors.toList());   
+    List<TimeRange> allBusyTimes = events.stream()
+      .filter(e -> eventContains(e.getAttendees(), request, true))
+      .map(e -> e.getWhen())
+      .collect(Collectors.toList()); 
```

Comment:
Interestingly here we would use +4 spaces: https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent because it is a continuation of the first line.